.github/workflows/tests.yml

```diff
@@ -77,15 +74,17 @@ jobs:
           fi
 
           echo "MIN_COVERAGE=$MIN_COVERAGE" >> "$GITHUB_OUTPUT"
-      - name: Run Unit tests
-        run: poetry run pytest -m "not azure and not functional" --junitxml=coverage-junit.xml --cov=. --cov-report xml:coverage.xml --cov-fail-under ${{ steps.coverage-value.outputs.MIN_COVERAGE }}
-      - uses: actions/upload-artifact@v4
-        if: ${{ !cancelled() }}
+
+      - name: Pre-build image and run unit tests with coverage in dev container
+        uses: devcontainers/ci@v0.3
         with:
-          name: coverage
-          path: |
-            coverage-junit.xml
-            coverage.xml
-          if-no-files-found: error
-      - name: Run Functional tests
-        run: poetry run pytest -m "functional"
+          imageName: ghcr.io/azure-samples/chat-with-your-data-solution-accelerator
+          runCmd: make unittest optional_args="--junitxml=coverage-junit.xml --cov=. --cov-report xml:coverage.xml --cov-fail-under ${{ steps.coverage-value.outputs.MIN_COVERAGE }}"
+          push: never
+
+      - name: Run other tests in dev container
+        uses: devcontainers/ci@v0.3
+        with:
+          imageName: ghcr.io/azure-samples/chat-with-your-data-solution-accelerator
+          runCmd: make ci
```

Comment:
Completely get why you've done this as I assume the frontend tests will not run without cypress installed. The original reason for the unit tests not to be in the devcontainer were down to speed. They can run in seconds rather than minutes (downloading the docker image).

Happy both ways as consistency is great! :-)

Reply:
I'm wondering if it's better to avoid using the `ci` task. If something fails now, you have to scan through a large amount of logs to see what went wrong. With each task separated into it's own step, you can easily see which part failed. You can also limit the logs to that failed task.

I'm wondering if it's worth removing/renaming the `ci` task. In the past I've used something like `check` to run all tests and linting locally, while having ci separated. Happy to have a call to discuss though!